company_from_url/models/ir_rule.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 AppsToGROW - Henrik Norlin
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import time
+from collections import defaultdict
+
+from odoo import api, fields, models, tools, SUPERUSER_ID, _
+from odoo.exceptions import ValidationError
+from odoo.osv import expression
+from odoo.tools.safe_eval import safe_eval
+
+import logging
+_logger = logging.getLogger(__name__)
+
+class IrRule(models.Model):
+    _inherit = 'ir.rule'
+
+    ''' COMPUTE DOMAIN PER COMPANY'''
+
+    # /base/ir/ir_rule.py
+    # This did not work:
+    # @api.model
+    # @tools.ormcache('self.env.uid', 'self.env.user.company_id', 'model_name', 'mode') #UPDATED
+    # def _compute_domain(self, model_name, mode="read"):
+    #     return super(IrRule, self)._compute_domain(model_name, mode=mode)
```

Comment:
@norlinhenrik I don't understand enough about `ormcache` to know how to make the above example work, but I think it's still worth trying. Maybe someone could assist us

Reply:
@hbrunn If you have time: my technical question here would be how to successfully override a value for `tools.ormcache` that was set on the parent function, in the child function. I have studied the code in `cache.py` in Odoo [link](https://github.com/OCA/OCB/blob/10.0/odoo/tools/cache.py#L33) and if I could somehow always clear the cache before the super() call, I think I would have solved my issue, ..... I'll keep on trying.

Reply:
I found something that [works](https://github.com/OCA/multi-company/pull/156/commits/2f316b5e986f5a0b767d8478519c8216072e59bc), but it doesnt seem very efficient - it often recomputes the cache, IMO too often.

Reply:
yes, you do this with every call, that's very inefficient, specifically because you clear all orm caches by clearing some ormcached function's cache if I read https://github.com/OCA/OCB/blob/10.0/odoo/tools/cache.py#L96 correctly. The net result then is that you have no caching at all for this function. Your specific case you can fix by just calling `_super._func_` (that's where the decorator lib that the cache uses puts a reference to the function the decorator wraps), as you want to completely circumvent the original caching. This can break if there's an override in between that's not decorated with the cache decorator at all, but you'll have to trust people don't break inheritance I guess.

To do this right, the framework should do the unwrapping when carving classes, but this also might be unwanted if we want super to be cached. Difficult.

Reply:
I have resorted to the below suggestion to clear caches when a user suddenly decides to use a different access domain. This seems to work.